### ISSUE TITLE
Document which Qt5 libraries are used

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#201]
We should document which Qt5 libraries are actually searched by our CMake scripts.

**Comment by lrineau** [comment]
On CGAL-develop, Joachim says:
```
for demo/Periodic_3_triangulation_3 I also need libqt5sql5-sqlite

Otherwise I get

[  0%] Generating Periodic_3_triangulation_3.qhc
QSqlDatabase: QSQLITE driver not loaded
QSqlDatabase: available drivers:
Cannot open data base file
/tmp/demo/Periodic_3_triangulation_3/Periodic_3_triangulation_3.qch.
Reading collection config file...
Generating help for Periodic_3_triangulation_3.qhp...
Building up file structure...
demo/Periodic_3_triangulation_3/CMakeFiles/periodic_3_triangulation_3_demo.dir/build.make:106:
recipe for target
'demo/Periodic_3_triangulation_3/Periodic_3_triangulation_3.qhc' failed
make[2]: *** [demo/Periodic_3_triangulation_3/Periodic_3_triangulation_3.qhc]
Error 255
make[2]: Target
'demo/Periodic_3_triangulation_3/CMakeFiles/periodic_3_triangulation_3_demo.dir/depend'
not remade because of errors.
CMakeFiles/Makefile2:3377: recipe for target
'demo/Periodic_3_triangulation_3/CMakeFiles/periodic_3_triangulation_3_demo.dir/all'
failed
make[1]: ***
[demo/Periodic_3_triangulation_3/CMakeFiles/periodic_3_triangulation_3_demo.dir/all]
Error 2
make[1]: Target 'demo/Periodic_3_triangulation_3/all' not remade because of errors.
Makefile:116: recipe for target 'all' failed
make: *** [all] Error 2
make: Target 'default_target' not remade because of errors.
```

Should we reopen this bug?

**Comment by sloriot** [comment]
Since there is nothing in the `CMakeList,txt` I wonder how libQt5Sqlit is linked with the demo.
I compiled it myself and the demo is also linked with libQt5CLucene.so.5 and libQt5Network.so.5

**Comment by sloriot** [comment]
They are coming with `libQt5Help.so`. Might be an error in the Debian package dependency?

**Comment by sloriot** [comment]
I just check and they are correctly picked up. Actually I don't understand why libqt5sql5-sqlite is needed. It just provide `/usr/lib/x86_64-linux-gnu/qt5/plugins/sqldrivers/libqsqlite.so` on my machine and it is not used by the demo.

**Comment by lrineau** [comment]
Actually the `sqllite` plugin of Qt5 is not something that is linked with. That is a plugin of Qt5. [It is loaded at runtime, if needed.](http://doc.qt.io/qt-5/sql-driver.html)

In https://github.com/CGAL/cgal/issues/201#issuecomment-128412950, the issue is that the tool `qcollectiongenerator` from Qt5 seems to require that plugin to be able to do its job. That is a packaging error from Debian: the package that provides the `qcollectiongenerator` should depend on `libqt5sql5-sqlite`.

**Comment by sloriot** [comment]
It's not an error I think on debian side. They are several plugins and you only need them depending on the input you want to read. There should be a way to handle that at the cmake level.

**Comment by sloriot** [comment]
See CGAL/CGAL-public-dev@156b267

**Comment by sloriot** [comment]
Fixed with CGAL/cgal-public-dev@999c123

**Comment by lrineau** [comment]
I think that is an error from Debian because the `qcollectiongenerator` tool is not usable without that plugin. Then the package providing `qcollectiongenerator` should require the package providing the plugin.
